### ISSUE TITLE
False positives for NUnit1003 when using enum values

### ISSUE THREAD

**Issue by seanm-fathomhq** [nunit/nunit.analyzers#552]
When using `TestCase` and one of the parameters is an `enum`, the analyser appears to think there isn't any parameters being passed in:
![image](https://github.com/nunit/nunit.analyzers/assets/108652239/b36c9882-033c-4c97-96e3-1e2eab362563)

If there are any other parameters, the analyser still sees 0:
![image](https://github.com/nunit/nunit.analyzers/assets/108652239/55766541-6a39-4214-b693-1ed9d00ba786)

If the enum value is optional, any `TestCase` with the enum value removed works as expected, but the cases with the enum added still report 0 parameters:
![image](https://github.com/nunit/nunit.analyzers/assets/108652239/c29b03d2-6593-4738-800f-966a477d1386)

**Comment by mikkelbu** [comment]
Hi @seanm-fathomhq

Which version of NUnit and which version of NUnit.Analyzers are you using? I cannot repro this, so can you provide a small and complete example that shows the problem - preferably as text, so we can easily test it.

When I try (NUnit 3.13.3 and NUnit.Analyzers 3.61) the following
```csharp
namespace TestTestCases
{
    public class TestCaseTest
    {
        [Test]
        [TestCase(SomeEnum.SomeValue)]
        [TestCase(SomeEnum.Undefined, SomeEnum.AnotherValue)]
        public void Test(SomeEnum test)
        {
            Assert.That(test, Is.Not.EqualTo(SomeEnum.SomeValue));
        }
    }

    public enum SomeEnum
    {
        Undefined,
        SomeValue,
        AnotherValue
    }
}
```
Then I get an error on the second case `NUnit1004 - The TestCaseAttribute provided too many arguments. Expected '1', but got '2'.`
![image](https://github.com/nunit/nunit.analyzers/assets/1007631/ab0f69be-2fb8-470b-8f3d-1bb30bd0645d)

**Comment by seanm-fathomhq** [comment]
Disregard 😩 I'm guessing it's an issue with my IDE or something, even trying to reproduce in the same file doesn't work today. If I ever manage to get it to reproduce cleanly and consistently I'll relog

**Comment by ci-vamp** [comment]
i was having the same issue after changing branches. i quit and reopened rider and the error went away.
